src/app/admin/new-book/new-book.component.spec.ts

```diff
@@ -127,24 +185,106 @@ describe('NewBookComponent', () => {
   });
 
   it('should call the uploadPicture method', async () => {
-    spyOn(component, 'uploadPicture');
-    el = fixture.debugElement.query(By.css('input')).nativeElement;
+    const spy = spyOn(component, 'uploadPicture').and.callThrough();
+    el = fixture.debugElement.query(By.css('#customFile')).nativeElement;
     el.click();
```

Comment:
You should prevent this from open up the user's file explorer while executing these tests.

Reply:
Fixed. Updated all occurrences of fixture.debugElement that was testing a method invoking via UI element event trigger